app/controllers/users_controller.rb

```diff
@@ -1,9 +1,19 @@
 class UsersController < ApplicationController
+  before_action :set_user, only: [:show]
+
   def index
-    # logic to fetch all users goes here
+    @users = User.all
   end
 
   def show
-    # logic to fetch a single user goes here
+    # @user.increment!(:posts_counter)
+  end
```

Comment:
- [ ] It is required to have show action in your user controller because that's what you're going to use in the user view show page to display user's 3 most recent post with the help of `#recent_post` method you created in the model. It should look something similar to this. :point_down: 

```
def show
  @user = User.find(params[:id])
end

```